src/controller/AutoCommissioner.cpp

```diff
@@ -262,12 +262,35 @@ CommissioningStage AutoCommissioner::GetNextCommissioningStageInternal(Commissio
         }
         else
         {
+            setCASEFailsafeTimerIfNeeded();
             if (mParams.GetSkipCommissioningComplete().ValueOr(false))
             {
                 return CommissioningStage::kCleanup;
             }
             return CommissioningStage::kFindOperational;
         }
+    case CommissioningStage::kScanNetworks:
+        return CommissioningStage::kNeedsNetworkCreds;
+    case CommissioningStage::kNeedsNetworkCreds:
+        if (mParams.GetWiFiCredentials().HasValue() && mDeviceCommissioningInfo.network.wifi.endpoint != kInvalidEndpointId)
```

Comment:
So this is now duplicating logic from the kSendNOC case.  Shouldn't kSendNOC just go to kNeedsNetworkCreds  if it does not go to kScanNetworks instead of duplicating this code?

Reply:
created helper function